src/api/c/handle.hpp

```diff
@@ -28,6 +28,8 @@ static const detail::Array<T> &
 getArray(const af_array &arr)
 {
     detail::Array<T> *A = reinterpret_cast<detail::Array<T>*>(arr);
+    if ((af_dtype)af::dtype_traits<T>::af_type != A->getType())
+        AF_ERROR("Internal error - contact developers.", AF_ERR_INTERNAL);
```

Comment:
Good check 👍 , but change the error to be "Expected foo got bar" kind of message here.